plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php

```diff
@@ -90,6 +104,23 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 		}
 		// TODO: If an image is visible in one breakpoint but not another, add loading=lazy AND add a regular-priority preload link with media queries (unless LCP in which case it should already have a fetchpriority=high link) so that the image won't be eagerly-loaded for viewports on which it is not shown.
 
+		// Ensure that sizes=auto is set properly.
+		$sizes = $processor->get_attribute( 'sizes' );
+		if ( is_string( $sizes ) ) {
+			$is_lazy  = 'lazy' === $get_attribute_value( 'loading' );
+			$has_auto = $this->sizes_attribute_includes_valid_auto( $sizes );
+
+			if ( $is_lazy && ! $has_auto ) {
+				$processor->set_attribute( 'sizes', "auto, $sizes" );
```

Comment:
@felixarntz If I understand you https://github.com/WordPress/performance/pull/1476#issuecomment-2297125561 correctly, are you thinking that this condition should be removed? IMO, including auto here would help ensure that the correct image _size_ is prioritized, so I think it makes sense in the scope of Image Prioritizer, especially if `sizes=auto` is merged into WordPress core.

Reply:
Doesn't have to be removed, but my point was that it's not strictly in scope of the Image Prioritizer plugin. To clarify, strictly speaking IMO whether this is done should be based on whether `sizes="auto"` is _generally_ supported:
* If it's available in Core on the site or another plugin on the site adds it, it makes sense that it's supported here too. That's because it would just maintain/correct Core's (or the other plugin's) behavior without making it worse (which would be _not_ having `sizes="auto"` on lazy-loaded images).
* But with current WordPress core and without another plugin doing it, it doesn't belong in this plugin. That's because it would effectively add this feature without it being its responsibility.

Now more pragmatically speaking, it probably doesn't matter all that much. Since we are on the path to adding `sizes="auto"` into Core already, we might as well add this here unconditionally, since it has no negative side effects and it would probably be unrealistic to reliably check whether _something else_ in WordPress generally adds `sizes="auto"`.

Hope this all makes sense. 😆